### ISSUE TITLE
unable to start krakend ion fedora 32

### ISSUE THREAD

**Issue by amarktl** [devopsfaith/krakend-ce#233]
starting krakend fails on fedora 32

i tried to install the generic binary as well as the rpm version and both fail with the same error.

### error message
```
krakend run -c /home/marktl/work/01_REQUEST/2020/api-gateway/sample-krakend.json
Parsing configuration file: /home/marktl/work/01_REQUEST/2020/api-gateway/sample-krakend.json
2020/07/17 10:40:08  ERROR: unable to create the gologging logger: getting the extra config for the krakend-gologging module
2020/07/17 10:40:08  ERROR: unable to create the GELF writer: getting the extra config for the krakend-gelf module
2020/07/17 10:40:08  INFO: Listening on port: 8080
2020/07/17 10:40:08  DEBUG: creating a new influxdb client
2020/07/17 10:40:08  DEBUG: no config for the influxdb client. Aborting
2020/07/17 10:40:08  WARNING: influxdb: unable to load custom config
2020/07/17 10:40:08  WARNING: opencensus: no extra config defined for the opencensus module
2020/07/17 10:40:08  WARNING: building the etcd client: unable to create the etcd client: no config
2020/07/17 10:40:08  DEBUG: no config for the bloomfilter
2020/07/17 10:40:08  WARNING: bloomFilter: no config for the bloomfilter
2020/07/17 10:40:08  WARNING: no config present for the httpsecure module
2020/07/17 10:40:08  INFO: registering usage stats for cluster ID '532pof80FRy80efZBaDQVc6FInYGoYLPGszsVYzG/C4='
2020/07/17 10:40:08  DEBUG: lua: no extra config
2020/07/17 10:40:08  DEBUG: botdetector middleware:  no config defined for the module
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/devopsfaith/krakend-amqp.backendFactory.initProducer(0x2576740, 0xc0006260c0, 0x2582e20, 0xc00020e5e0, 0xc000599ad0, 0xc0005113b0, 0xc000419610, 0x2576740, 0xc0006260c0, 0xc000628500, ...)
	/go/pkg/mod/github.com/devopsfaith/krakend-amqp@v1.1.0/producer.go:46 +0x86c
github.com/devopsfaith/krakend-amqp.backendFactory.New(0x2576740, 0xc0006260c0, 0x2582e20, 0xc00020e5e0, 0xc000599ad0, 0xc0005113b0, 0xc000419610, 0xc000628500, 0xc00093ea60)
	/go/pkg/mod/github.com/devopsfaith/krakend-amqp@v1.1.0/client.go:39 +0xe4
github.com/devopsfaith/krakend-lambda.BackendFactoryWithInvoker.func1(0xc000628500, 0x0)
	/go/pkg/mod/github.com/devopsfaith/krakend-lambda@v0.0.0-20200123123306-cce9724ca461/backend.go:51 +0x169
github.com/devopsfaith/krakend-cel.BackendFactory.func1(0xc000628500, 0xc000100000)
	/go/pkg/mod/github.com/devopsfaith/krakend-cel@v0.0.0-20190502130550-d6872fd4f97e/proxy.go:41 +0x59
github.com/devopsfaith/krakend-lua/proxy.BackendFactory.func1(0xc000628500, 0x22b5e30)
	/go/pkg/mod/github.com/devopsfaith/krakend-lua@v1.1.0/proxy/proxy.go:38 +0x5f
github.com/devopsfaith/krakend-ratelimit/juju/proxy.BackendFactory.func1(0xc000628500, 0x2582e20)
	/go/pkg/mod/github.com/devopsfaith/krakend-ratelimit@v1.1.0/juju/proxy/proxy.go:59 +0x55
github.com/devopsfaith/krakend-circuitbreaker/gobreaker/proxy.BackendFactory.func1(0xc000628500, 0x8)
	/go/pkg/mod/github.com/devopsfaith/krakend-circuitbreaker@v0.0.0-20190405155253-cf7c0a538981/gobreaker/proxy/proxy.go:38 +0x67
github.com/devopsfaith/krakend/proxy.defaultFactory.newStack(0xc00020ece0, 0x2582e20, 0xc00020e5e0, 0x22b5ee8, 0xc000628500, 0x18)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:84 +0x34
github.com/devopsfaith/krakend/proxy.defaultFactory.newSingle(...)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:80
github.com/devopsfaith/krakend/proxy.defaultFactory.New(0xc00020ece0, 0x2582e20, 0xc00020e5e0, 0x22b5ee8, 0xc0001ac640, 0x22b8c48, 0xc000624660, 0x1e65560)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:58 +0x15a
github.com/devopsfaith/krakend/proxy.shadowFactory.New(0x25431c0, 0xc00020ed00, 0xc0001ac640, 0x225b6dd, 0x2b, 0xc00093ef88)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/shadow.go:38 +0x363
github.com/devopsfaith/krakend-jsonschema.ProxyFactory.func1(0xc0001ac640, 0x1f53ea0, 0xc00087c1b0, 0xa2900000076)
	/go/pkg/mod/github.com/devopsfaith/krakend-jsonschema@v0.0.0-20190124184701-5705a5015d7a/jsonschema.go:25 +0x3d
github.com/devopsfaith/krakend/proxy.FactoryFunc.New(0xc00020ed20, 0xc0001ac640, 0xc00005a000, 0xc00093f020, 0xd0a77f)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:23 +0x30
github.com/devopsfaith/krakend-cel.ProxyFactory.func1(0xc0001ac640, 0xc00093f0b0, 0xda0902, 0xc000134060)
	/go/pkg/mod/github.com/devopsfaith/krakend-cel@v0.0.0-20190502130550-d6872fd4f97e/proxy.go:17 +0x63
github.com/devopsfaith/krakend/proxy.FactoryFunc.New(0xc000511470, 0xc0001ac640, 0xc000134060, 0xc000130200, 0x56)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:23 +0x30
github.com/devopsfaith/krakend-lua/proxy.ProxyFactory.func1(0xc0001ac640, 0xc000130200, 0x56, 0x80)
	/go/pkg/mod/github.com/devopsfaith/krakend-lua@v1.1.0/proxy/proxy.go:21 +0x69
github.com/devopsfaith/krakend/proxy.FactoryFunc.New(0xc000511740, 0xc0001ac640, 0xc000599b40, 0x10, 0x10)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:23 +0x30
github.com/devopsfaith/krakend/proxy.FactoryFunc.New(0xc00020ed40, 0xc0001ac640, 0xcdf55a, 0xc000599b40, 0xc000599a40)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:23 +0x30
github.com/devopsfaith/krakend/proxy.FactoryFunc.New(0xc00020ed60, 0xc0001ac640, 0x1f25100, 0xc000599b40, 0x0)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/proxy/factory.go:23 +0x30
github.com/devopsfaith/krakend/router/gin.ginRouter.registerKrakendEndpoints(0xc0002a2140, 0x0, 0x0, 0x0, 0xc00020ede0, 0x25431a0, 0xc00020ed60, 0x2582e20, 0xc00020e5e0, 0xc00020ee00, ...)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/router/gin/router.go:105 +0x8b
github.com/devopsfaith/krakend/router/gin.ginRouter.Run(0xc0002a2140, 0x0, 0x0, 0x0, 0xc00020ede0, 0x25431a0, 0xc00020ed60, 0x2582e20, 0xc00020e5e0, 0xc00020ee00, ...)
	/go/pkg/mod/github.com/devopsfaith/krakend@v1.1.1/router/gin/router.go:90 +0x19f
github.com/devopsfaith/krakend-ce.(*ExecutorBuilder).NewCmdExecutor.func1(0xc000596394, 0xa, 0xc000624020, 0x1, 0x1, 0xb2d05e00, 0x45d964b800, 0x3393bf0, 0x0, 0x0, ...)
	/app/executor.go:158 +0x755
github.com/devopsfaith/krakend-cobra.runFunc(0x32452e0, 0xc00000e280, 0x0, 0x2)
	/go/pkg/mod/github.com/devopsfaith/krakend-cobra@v0.0.0-20190403130617-3f056666a23e/run.go:24 +0x189
github.com/spf13/cobra.(*Command).execute(0x32452e0, 0xc00000e1e0, 0x2, 0x2, 0x32452e0, 0xc00000e1e0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x3244e20, 0xcd8188, 0x20, 0x20282e0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ea
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
github.com/devopsfaith/krakend-cobra.Execute(0x2543160, 0xc000728000, 0xc00000e140)
	/go/pkg/mod/github.com/devopsfaith/krakend-cobra@v0.0.0-20190403130617-3f056666a23e/cmd.go:20 +0x62
main.main()
	/app/cmd/krakend-ce/main.go:56 +0x359
```

### config file

```
{
  "version": 2,
  "extra_config": {},
  "timeout": "3000ms",
  "cache_ttl": "300s",
  "output_encoding": "json",
  "name": "local-echo",
  "endpoints": [
    {
      "endpoint": "/echo",
      "method": "GET",
      "output_encoding": "json",
      "extra_config": {},
      "querystring_params": [
        "message"
      ],
      "backend": [
        {
          "url_pattern": "/sample",
          "encoding": "json",
          "sd": "static",
          "method": "GET",
          "extra_config": {}
        }
      ]
    }
  ]
}
```

### system information

```
uname -a
Linux  5.7.8-200.fc32.x86_64 #1 SMP Thu Jul 9 14:34:51 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
lscpu
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          8
On-line CPU(s) list:             0-7
Thread(s) per core:              2
Core(s) per socket:              4
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           94
Model name:                      Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
Stepping:                        3
CPU MHz:                         2700.030
CPU max MHz:                     3600.0000
CPU min MHz:                     800.0000
BogoMIPS:                        5399.81
Virtualization:                  VT-x
L1d cache:                       128 KiB
L1i cache:                       128 KiB
L2 cache:                        1 MiB
L3 cache:                        8 MiB
NUMA node0 CPU(s):               0-7
Vulnerability Itlb multihit:     KVM: Mitigation: Split huge pages
Vulnerability L1tf:              Mitigation; PTE Inversion; VMX conditional cache flushes, SMT vulnerable
Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Full generic retpoline, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling
Vulnerability Srbds:             Vulnerable: No microcode
Vulnerability Tsx async abort:   Mitigation; Clear CPU buffers; SMT vulnerable
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xto
                                 pology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lah
                                 f_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx rdseed adx smap 
                                 clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d
```

**Comment by kpacha** [comment]
it should be fixed at 1.1.1, after this PR was included: https://github.com/devopsfaith/krakend-amqp/pull/5

**Comment by amarktl** [comment]
after adding a proper host configuration everything runs fine. the main problem for me was that the configuration check reported it as valid.
